src/content_sao.h

```diff
@@ -42,9 +40,51 @@ class UnitSAO: public ServerActiveObject
 	s16 getHP() const { return m_hp; }
 	// Use a function, if isDead can be defined by other conditions
 	bool isDead() const { return m_hp == 0; }
+
+	void setArmorGroups(const ItemGroupList &armor_groups);
+	ItemGroupList getArmorGroups();
+	virtual std::string getPropertyPacket();
+	ObjectProperties* accessObjectProperties();
+	void notifyObjectPropertiesModified();
+	void setAnimation(v2f frame_range, float frame_speed, float frame_blend, bool frame_loop);
+	void getAnimation(v2f *frame_range, float *frame_speed, float *frame_blend, bool *frame_loop);
+	void setBonePosition(const std::string &bone, v3f position, v3f rotation);
+	void getBonePosition(const std::string &bone, v3f *position, v3f *rotation);
+	void setAttachment(int parent_id, const std::string &bone, v3f position, v3f rotation);
+	void getAttachment(int *parent_id, std::string *bone, v3f *position, v3f *rotation);
+	void addAttachmentChild(int child_id);
+	void removeAttachmentChild(int child_id);
+	UNORDERED_SET<int> getAttachmentChildIds();
```

Comment:
if we didn't write on the result, could be very good to use const ref on the result